gateway/utils/metrics/projects.go

```diff
@@ -0,0 +1,125 @@
+package metrics
+
+import (
+	"context"
+	"github.com/sirupsen/logrus"
+	"github.com/spaceuptech/space-api-go/types"
+	"github.com/spaceuptech/space-cloud/gateway/utils"
+	"net/http"
+	"runtime"
+	"time"
+)
+
+func currentTimeInMillis() int64 {
+	// subtracting interval time make sures that multiple gateways in a cluster don't write to database frequently
+	return time.Now().Add(-metricsUpdaterInterval).UnixNano()
+}
+
+func (m *Module) generateMetricsRequest() (string, map[string]interface{}, map[string]interface{}, bool) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	c := m.syncMan.GetGlobalConfig()
+	if c == nil {
+		return "", nil, nil, true
+	}
+
+	// Get the cluster size
+	clusterSize, err := m.syncMan.GetClusterSize(context.Background())
+	if err != nil {
+		clusterSize = 1
+	}
+	find := m.syncMan.GetClusterID()
```

Comment:
Shouldn't find be a map[string]interface{}

Reply:
not needed, update scMetrics handles it